### PR TITLE
chore: added test setup for react-query and integrated react-query

### PR DESCRIPTION
**Pull request by Jscripter-pk**

### CODE REVIEW

**Comment by Jscripter-pk** [comment]
@aziyatali you can move forward from here for adding functions. As I have added setup for test and integrations for react-query in this PR

File: src/kross-client/base.ts [hunk 1]
```diff
@@ -62,13 +71,15 @@ export class KrossClientBase {
...
+  login({ keyid, password }: LoginDto) {
+    return this.instance.post<LoginResponse>('/auth/login', {
       keyid,
       password,
     });
-    this.authToken = res.data.token;
-    this.refreshToken = res.data.refresh;
+  }
+
+  async accountData() {
```
Comment (JonghunYu) [inline src/kross-client/base.ts:81]: I think we should add this functino to account module

File: src/kross-client/base.ts [hunk 2]
```diff
@@ -80,6 +91,31 @@ export class KrossClientBase {
...
+  useAccountHooks() {
+    return {
+      accountData: () => {
+        return useQuery({
+          queryKey: 'accountData',
+          queryFn: () => this.accountData.bind(this),
+        });
+      },
+    };
+  }
```
Comment (JonghunYu) [inline src/kross-client/base.ts:103]: this should be in account too

**Review by JonghunYu — APPROVED** [review]
@Jscripter-pk Please check the comments

**Review by JonghunYu — APPROVED** [review]

**Comment by Jscripter-pk** [comment]
@JonghunYu @aziyatali Why did you guys closed this?

**Comment by aziyatali** [comment]
@Jscripter-pk  I am going to add them separately category by category with testing as well such as on https://olive-dev.kross.kr/docs

**Pull request merged**